HomeWork2/src/index.ts

```diff
@@ -0,0 +1,182 @@
+class School {
+  // implement 'add area', 'remove area', 'add lecturer', and 'remove lecturer' methods
+
+  _areas = [];
+  _lecturers: string[] = []; // Name, surname, position, company, experience, courses, contacts
+
+  get areas(): any[] {
+    return this._areas;
+  }
+
+  get lecturers(): string[] {
+    return this._lecturers;
+  }
+
+  addArea(area: string): void {
+    this._lecturers.push(area);
+  }
+
+  removeArea(area: string): void {
+    this._lecturers = this._lecturers.filter(value => value !== area);
+  }
+
+  addLecturer(lacturer: string): void {
+    this._lecturers.push(lacturer);
+  }
+
+  removeLecturer(lacturer: string): void {
+    this._lecturers = this._lecturers.filter(value => value !== lacturer);
+  }
+}
+
+class Area {
+  // implement getters for fields and 'add/remove level' methods
+  _levels: string[] = [];
+  _name: string;
+
+  constructor(name: string) {
+    this._name = name;
+  }
+
+  get levels(): string[] {
+    return this._levels;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+
+  addLevel(level: string): void {
+    this._levels.push(level);
+  }
+
+  removeLevel(level: string) {
+    this._levels = this._levels.filter(valueLevel => valueLevel !== level);
+  }
+}
+
+class Level {
+  // implement getters for fields and 'add/remove group' methods
+
+  _groups: string[] = [];
+  _name: string;
+  _description: string;
+
+  constructor(name: string, description: string) {
+    this._name = name;
+    this._description = description;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+
+  get description(): string {
+    return this._description;
+  }
+
+  get groups(): string[] {
+    return this._groups;
+  }
+
+  addGroup(group: string): void {
+    this._groups.push(group);
+  }
+
+  removeGroup(group: string): void {
+    this._groups = this._groups.filter(valueGroup => valueGroup !== group);
+  }
+}
+
+class Group {
+  // implement getters for fields and 'add/remove student' and 'set status' methods
+
+  _area: string = '';
+  _status: string = '';
+  _students: any[] = []; // Modify the array so that it has a valid toSorted method*
```

Comment:
Type `any` is not acceptable from now for us. Please, specify object annotation to cover this case.